build-system/pr-check/utils.js

```diff
@@ -0,0 +1,205 @@
+  const startTime = Date.now();
+  const fileLogPrefix = colors.bold(colors.yellow(`${fileName}:`));
+  console.log(
+      '\n' + fileLogPrefix, 'Running', colors.cyan(functionName) + '...');
+  return startTime;
+}
+
+/**
+ * Stops the timer for the given function and prints the execution time.
+ * @param {string} functionName
+ * @param {string} fileName
+ * @param {DOMHighResTimeStamp} startTime
+ * @return {number}
+ */
+function stopTimer(functionName, fileName, startTime) {
+  const endTime = Date.now();
+  const executionTime = new Date(endTime - startTime);
+  const mins = executionTime.getMinutes();
+  const secs = executionTime.getSeconds();
+  const fileLogPrefix = colors.bold(colors.yellow(`${fileName}:`));
+  console.log(
+      fileLogPrefix, 'Done running', colors.cyan(functionName),
+      'Total time:', colors.green(mins + 'm ' + secs + 's'));
+}
+
+/**
+ * Executes the provided command and times it. Errors, if any, are printed.
+ * @param {string} cmd
+ * @return {<Object>} Process info.
+ */
+function timedExec(cmd) {
+  const startTime = startTimer(cmd);
+  const p = exec(cmd);
+  stopTimer(cmd, startTime);
+  return p;
+}
+
+/**
+ * Executes the provided command and times it. The program terminates in case of
+ * failure.
+ * @param {string} cmd
+ * @param {string} fileName
+ */
+function timedExecOrDie(cmd, fileName = 'utils.js') {
+  const startTime = startTimer(cmd, fileName);
+  execOrDie(cmd);
+  stopTimer(cmd, fileName, startTime);
+}
+
+/**
+ * Downloads build output from storage
+ * @param {string} functionName
+ */
+function downloadBuildOutput(functionName) {
+  const fileLogPrefix = colors.bold(colors.yellow(`${functionName}:`));
+  const buildOutputDownloadUrl =
+    `${BUILD_OUTPUT_STORAGE_LOCATION}/${BUILD_OUTPUT_FILE}`;
+
+  console.log(
+      `${fileLogPrefix} Downloading build output from ` +
+      `${colors.cyan(buildOutputDownloadUrl)}...`);
+  exec('echo travis_fold:start:download_results');
+  execOrDie(`gsutil cp ${buildOutputDownloadUrl} ${BUILD_OUTPUT_FILE}`);
+  exec('echo travis_fold:end:download_results');
+
+  console.log(
+      `${fileLogPrefix} Extracting ${colors.cyan(BUILD_OUTPUT_FILE)}...`);
+  exec('echo travis_fold:start:unzip_results');
+  execOrDie(`unzip -o ${BUILD_OUTPUT_FILE}`);
+  exec('echo travis_fold:end:unzip_results');
+
+  console.log(fileLogPrefix, 'Verifying extracted files...');
+  exec('echo travis_fold:start:verify_unzip_results');
+  execOrDie(`ls -la ${BUILD_OUTPUT_DIRS}`);
+  exec('echo travis_fold:end:verify_unzip_results');
+}
+
+/**
+ * Zips and uploads the build output to a remote storage location
+ * @param {string} functionName
+ */
+function uploadBuildOutput(functionName) {
+  const fileLogPrefix = colors.bold(colors.yellow(`${functionName}:`));
+
+  console.log(
+      `\n${fileLogPrefix} Compressing contents of directories ` +
+      `${colors.cyan(BUILD_OUTPUT_DIRS)} into ` +
+      `${colors.cyan(BUILD_OUTPUT_FILE)}...`);
```

Comment:
Style nit: Don't include color formatting inside `${}`. Prefer `cyan(${foo})` to `${cyan(foo)}`, here and elsewhere in the PR.

Also, on reading the logs, there are files and directories being compressed, so let's rewrite this line as:
```
  const outputDirList = BUILD_OUTPUT_DIRS.split(' ').join(',');
  console.log(
      `\n${fileLogPrefix} Compressing ` +
      `colors.cyan(${outputDirList}) into ` +
      `colors.cyan(${BUILD_OUTPUT_FILE})...`);
```

(See https://travis-ci.org/ampproject/amphtml/jobs/493824868#L751)